### DISCUSSION TITLE
estimate_uc_dc_atten function outputs are giving strange results for the rest of tuning

### DISCUSSION THREAD

**Discussion by prakamya-agrawal** [simonsobs/sodetlib#357] [Issues and Debugging]
I ran `estimate_uc_dc_atten` on all bands in the most recent tuning of Mv4 at SLAC. I previously tried optimizing UC and DC attenuation by hand and wanted to compare my results to the function. 

**My results:**
<img width="400" alt="image" src="https://github.com/simonsobs/sodetlib/assets/37946975/b80c6278-f221-4d04-938c-f2d159ffe06e">

**Results with `estimate_uc_dc_atten`:**
![image](https://github.com/simonsobs/sodetlib/assets/37946975/2c24e44a-3306-48df-b91c-6df655704878)

This resulted in `estimate_phase_delay ` and `setup_tune` looking weird for bands 0-3:
![image](https://github.com/simonsobs/sodetlib/assets/37946975/d9459af1-4ca1-4c23-a963-7b2074a8a5aa)
![image](https://github.com/simonsobs/sodetlib/assets/37946975/2a80d48e-dd3e-46ed-b3ab-c9dfcc7a6047)

Not sure if this is an actual bug with the function or anything but I thought it was noteworthy, and I'm confused as to why it gave me this result

**Comment by jlashner** [comment]
That find-freq response looks kind of like a set of amplifiers isn't powered correctly. If amplifiers are on, you should still be able to see the band even if the attens aren't set perfectly, and it makes sense that estimate-phase-delay and estimate-attens aren't working properly if the amplifiers aren't on.

**Reply by prakamya-agrawal** [reply]
Yeah, I set the amplifiers before running anything
<img width="277" alt="image" src="https://github.com/simonsobs/sodetlib/assets/37946975/cec5e0e2-eee6-4a5a-b2bc-66d80fdedfd6">

**Reply by jlashner** [reply]
hemt1 and 50k1 drain currents definitely look off here.

**Reply by msilvafe** [reply]
I think their 50k1 amp is a different type of amp B&Z tech which only takes a drain voltage and no gate voltage so I think that one reads fine to me and their hemt1 is also specified 0.5V/4mA on their datasheet so it doesn't seem too far off. The other amplifier voltages and currents also match but looking on the SLAC confluence I see a note for hemt2: "Strongly suspect either this amplifier or the LNF amplifier it was used with at Pole are broken ; see glitching in gain in Pickles in early cooldowns, which might also be an issue we had at Pole." Do you know of this @Prakamya01 and/or is the bad band you show above connected to hemt2?

**Reply by msilvafe** [reply]
But I agree w/ Jack that it seems like there's no transmission, do all of the bands on this AMC look like that? If so maybe your 300K amplifier on that chain isn't biased?

**Reply by prakamya-agrawal** [reply]
1. I didn't know of the hemt2 note. I feel like every time we've used it it's been very stable so idk
2. yeah all of the bands on AMC0 look like that. how do i check the 300K amp?

**Reply by jlashner** [reply]
This looks to me like a failure of the HEMT rather than the 300K, I think when the 300K is malfunctioning we can still see _some_ transmission, even if the amplitude is much smaller. To check, maybe you can try turning off one amplifier at a time, and comparing   the results of `full_band_amplitude_sweep` for each state. If you do something like this, you would see that the response when the broken amplifier is off is roughly the same as the response when smurf thinks all three amplifiers are on. 

It would be awesome if you could post plots from that here, so we know what the transmission looks like when any of the amplifiers have failed.

**Reply by prakamya-agrawal** [reply]
I ran `estimate_uc_dc_atten`, and then I went one by one and turned off one amplifier, leaving the others on, and ran `full_band_resp` (I wasn't sure at what tone power  or subband to run `full_band_ampl_sweep`). Here are the results for when each amplifier is off. Not sure how to interpret it. 
[full_band_resp with 1 amplifier off at a time.pdf](https://github.com/simonsobs/sodetlib/files/11810758/full_band_resp.with.1.amplifier.off.at.a.time.pdf)

**Comment by jlashner** [comment]
Hi Prakamya, were you ever able to get this to work? Honestly the plots that you sent above look fine to me, like whenever both amps were on for both AMC0 and AMC1, the transmitted signals look like I expect. If you enable both pairs of amps and run find_freq again, do you still get that it looks like there's no transmission?

**Reply by prakamya-agrawal** [reply]
I didn't try running find_freq again, but I think we're just going to set the uc and dc attenuation manually.

**Reply by jlashner** [reply]
Ok cool, lmk if find_freq ends up working.
